Refactor num cpu

valentin-krasontovitsch opened the pull request:
**Issue**
Closes #4005


## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

valentin-krasontovitsch commented:
jenkins retest this please

eivindjahren commented on tests/libres_tests/res/enkf/test_subst_config.py:
```diff
@@ -17,23 +17,22 @@ def without_key(a_dict, key):
         ConfigKeys.CONFIG_FILE_KEY: "config",
         ConfigKeys.DEFINE_KEY: {"keyA": "valA", "keyB": "valB"},
         ConfigKeys.DATA_KW_KEY: {"keyC": "valC", "keyD": "valD"},
-        ConfigKeys.DATA_FILE: "eclipse/model/SNAKE_OIL.DATA",
```
Is it really necessary to modify the test in this way?

valentin-krasontovitsch replied:
i think i didn't test if it works with it, or not - i think i removed it because subst config no longer uses data file from the config.
so you might call it misleading to keep it in there, or just non minimalistic.
if you want, i can put in a separate case that tests whether we get the expected behaviour when DATA_FILE is present - is this desired?

eivindjahren replied:
It would be nice to know why the test fails without removing `ConfigKeys.DATA_FILE` and then add some testing for the new behavior.

valentin-krasontovitsch replied:
but but but... i don't think it's failing...
i can try to see what happens when i add it back

valentin-krasontovitsch replied:
as i didn't remember, but intuitively suspected ; ) the test doth not faileth when one adds the DATA FILE - it has no effect, as should be expected. so i think we can just leave it out. i'm gonna resolve this.

feel free to merge yourself, whenever you approve!

eivindjahren reviewed (commented):
I just had a question about the need to change one test, otherwise I see no reason to postpone merging :)

valentin-krasontovitsch commented:
comment resolved - but i have a comment myself. please, any of the reviewers, have a look at commit https://github.com/equinor/ert/pull/4056/commits/0e802c4def0fcb925c0e5cb9225f338c517113a2 specifically - i don't really know if we want to merge that one. feel free to just drop it.

also, remember to fixup / squash the commits and meld the (meaningful parts of the) commit messages before merging!

eivindjahren commented:
> comment resolved - but i have a comment myself. please, any of the reviewers, have a look at commit [0e802c4](https://github.com/equinor/ert/commit/0e802c4def0fcb925c0e5cb9225f338c517113a2) specifically - i don't really know if we want to merge that one. feel free to just drop it.
> 
> also, remember to fixup / squash the commits and meld the (meaningful parts of the) commit messages before merging!

Please dont have TODOs in commit messages.

valentin-krasontovitsch commented:
closing in favor of #4101

valentin-krasontovitsch commented:
reopening as i actually favor this one after all.

rebased unto main, removed unused ecl function, and removed num cpu from queue config as its not used at all!

valentin-krasontovitsch commented:
apparently i cannot reopen...

oysteoh commented:
If you hover the reopen button it states this one cannot be reopened due to it already exists an open PR on this branch.
